packages/flutter_tools/lib/src/crash_reporting.dart

```diff
@@ -133,4 +151,183 @@ class CrashReportSender {
+    printStatus('====================================================================');
+    printStatus('');
+
+    bool always = config.getValue('engine-crash-reporting'); // Read this value from disk in flutter settings.
+    if (!always) {
+      printStatus('Reporting this crash will help improve Flutter.', emphasis: true);
+      final String input = (await terminal.promptForCharInput(
+        <String>['y', 'n', 'a', 'Y', 'N', 'A'],
+        prompt: 'Would you like to send this Engine crash report to Google?\n([y]es|[N]o|[a]lways)',
+        defaultChoiceIndex: 1,
+        displayAcceptedCharacters: false)).toLowerCase().trim();
+      if (input == 'a' || input == 'always') {
+        // Set flag in settings
+        config.setValueBool('engine-crash-reporting', true);
+        always = true;
+      } else if (input == 'y' || input == 'yes') {
+        // Nothing to do.
+      } else if (input == 'n' || input == 'no' || input == '') {
+        // "No" or no recognized response. Default to not sending report.
+        printStatus('Skipped sending engine crash report.');
+        completer.complete();
+        return;
+      }
+    }
+    try {
+      printStatus('Sending crash report to Google.');
+      if (always) {
+        printStatus('===========================================================================');
+        printStatus('If you would like to stop automatically sending engine crash reports, use: ');
+        printStatus('  flutter config --no-engine-crash-reporting');
+        printStatus('===========================================================================');
+      }
+
+      final String flutterVersion = getFlutterVersion();
+      final Uri uri = _baseUrl.replace(
+        queryParameters: <String, String>{
+          'product': _kProductEngineId,
+          'version': flutterVersion,
+        },
+      );
+
+      final http.MultipartRequest req = http.MultipartRequest('POST', uri);
+      req.fields['uuid'] = _usage.clientId;
+      req.fields['product'] = _kProductEngineId;
+      req.fields['version'] = flutterVersion;
+      req.fields['osName'] = platform.operatingSystem;
+      req.fields['osVersion'] = os.name; // this actually includes version
+      req.fields['type'] = _kEngineTypeId;
+      req.fields['signature'] = engineCrash.parseSignature();
+      req.fields['error_runtime_type'] = _kEngineTypeId;
+
+      if (engineCrash.lineContaining('backtrace') != -1) {
+        req.files.add(http.MultipartFile.fromString(
+          _kEngineStackTraceFileField,
+          engineCrash.squashBacktrace(
+            startLine: engineCrash.lineContaining('backtrace') + 1,
+          ),
+          filename: _kEngineStackTraceFileField,
+        ));
+
+        req.files.add(http.MultipartFile.fromString(
+          _kEngineRegistersFileField,
+          engineCrash.squashBacktrace(
+            startLine: engineCrash.lineContaining('backtrace') - 5,
```

Comment:
The `engineCrash.lineContaining('backtrace') - 5` (and later ` - 4`) seems fragile. Could the format changes in the future? Could the index be out of range if the format changes? Here are a few things that I think could help:

1. Add a sample format (`EngineCrash._backtrace` lines) in the comment so if it crashes in the future, we at least know the expected format.
2. Do a more direct match/search for the field, instead of searching for another field, and rely on the relative position between the two fields.
3. Handle the failure more elegantly with helpful error message (e.g., instead of crash on accessing negative index, catch it and print something like "unexpected format..."), and fall back to your "report entire error" branch below.

Reply:
I think I will just directly return the engine error always. The backtrace/register parsing isn't particularly useful anyways, and the full error is not that long.